packages/catalog-model/src/kinds/ApiEntityV1alpha1.ts

```diff
@@ -31,7 +31,10 @@ export interface ApiEntityV1alpha1 extends Entity {
     type: string;
     lifecycle: string;
     owner: string;
-    definition: string;
+    // TODO: We need to adjust tests and documentation for this change.
+    // TODO: Think about a migration strategy!
+    definition?: string;
+    definitionLocation: string;
```

Comment:
Making the definition optional (only keeping them as an option for ingestion) is a breaking change I guess. In the future one could also use definitionLocation to reference API definitions, but that is not implemented yet. Haven't tried it yet, but it would be neat if one could write:

```yaml
spec:
  type: openapi
  definitionLocation: url:./openapi.yaml
```

Reply:
Not sure.

However, the longer I'm working with it, the less I like the idea of using a field that points to a location here. The location concept provides a bit too much flexibility, which makes it pretty hard to do assumptions in code.

Instead we could remove this indirection and let consumers directly work with the attachment keys. That would mean that the API entity kind has a notice that it uses the attachment `backstage.io/api-definition` to store API definitions. Code that requests this, directly fetches from this key, without having a `definitionLocation`.

What impact would that have? On the client side we can rely on always using the CatalogClient, without having to interpret the `definitionLocation`. If someone is interested to load them from a different service, they can still swap out the implementation in the backend, or accept that the backend is caching the data.
The bad side is, that we need additional details in the documentation of a kind, that describes the related attachments - there is also no static checking as the attachments are not part of the entity model. It's also the question how we can efficiently ingest definitions from other files: `definitionLocation` provides a nice pattern that would allow that, just embedding with `$json: ...` makes it difficult.

We could also keep `definitionLocation` only as an optional field in the backend used for ingestion and use a fixed key in the frontend.

I think that is something we should decide before moving on.